merkletree/pad.go

```diff
@@ -114,8 +114,9 @@ func (pad *PAD) GetSTR(epoch uint64) *SignedTreeRoot {
 func (pad *PAD) TB(key string, value []byte) (*TemporaryBinding, error) {
 	//FIXME: compute private index twice
 	//it would be refactored after merging VRF integration branch
-	index, _ := pad.computePrivateIndex(key)
-	tb := pad.currentSTR.sig
+	cStr := pad.currentSTR
```

Comment:
> also, in this case, when you do the lookup from a previous epoch, you still need to use the key from the old policy.

I think this is done in e82f666f1930a475550a8c3eb00194932c003a19 

> Then, when you call Update(policies), it should generate the STR with the policies that were actually used (pad.policies), and then swap the pad.policies with the ones being passed in to be used going forward. It can then reshuffle the pad.tree if the private key changed.

Besides the reshuffling part this is also done in the latest commit. Please review. I have some questions regarding the design of the reshuffling (below).